src/app/util/helper.ts

```diff
@@ -22,5 +22,9 @@ export const calculateSlippage = (base: string, percent: number | undefined = 0)
 };
 
 export const formatInputTokenValue = (base: number, exponent: string) => {
-  return base * Math.pow(10, parseFloat(exponent));
+  return Math.floor(base * Math.pow(10, parseFloat(exponent))).toString();
+};
+
+export const formatDecimalsFromToken = (base: number, decimals: string) => {
```

Comment:
Are u sure that this will work correctly in the future? JS has a problem with decimals sometimes. Maybe we should install Decimal npm and try to use it in all places.

Reply:
i went over this with Kaliman a couple of times, it should work fine, he told me that this is a simple function, if you want i can replace it with the other library anyway but i am not sure it is needed, we use it to add decimals to the number.